charms/reactive/relations.py

```diff
@@ -499,6 +524,31 @@ def remove_state(self, state):
         else:
             remove_state(state)
 
+    def is_state(self, state):
+        """
+        Test if this conversation is in the given state.
+        """
+        state = state.format(relation_name=self.relation_name)
+        value = get_state(state)
+        if not value:
+            return False
+        return self.key in value['conversations']
+
+    def toggle_state(self, state, active=None):
+        """
+        Toggle the given state for this conversation.
+
+        If ``active`` is not given, the state will be flipped from its current
```

Comment:
Flipped is a bit confusing here, I would explicitly call out the state will be removed or set.